### PR TITLE
added elas to adv-filters

### PR DESCRIPTION
**Pull request by dsnyder-fairway**
not using the filter
screen shot 2018-03-29 at 3 54 36 pm

using the filter
screen shot 2018-03-29 at 3 54 41 pm

### CODE REVIEW

File: report-processor/src/main/resources/application.yml [hunk 1]
```diff
@@ -146,3 +146,5 @@ reporting:
   state:
     code: CA
     name: California
+  # english-learners should be lep or elas
+  english-learners: lep
```
Comment (dsnyder-fairway) [inline report-processor/src/main/resources/application.yml]: keeping this at `lep` for now. Also, I would like a better name than `english-learners` but couldn't come up with anything better

**Comment by dsnyder-fairway** [comment]
Advanced Filters is at (or has already surpassed) the amount of stuff we can jam into one row. How should we break this up?

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettings.java [hunk 2]
```diff
@@ -20,6 +20,23 @@
...
+        this.englishLearners = englishLearners;
+    }
+
+    public boolean isLepEnabled() {
+        return "lep".equals(englishLearners);
+    }
+
+    public boolean isElasEnabled() {
+        return "elas".equals(englishLearners);
+    }
```
Comment (dsnyder-fairway) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettings.java]: this allows for some flexibility if there was ever a some need for both lep and elas. We make no assumptions on the front end that this is an XOR property

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettings.java [hunk 3]
```diff
@@ -20,6 +20,23 @@
...
+
+    public String getEnglishLearners() {
+        return englishLearners;
+    }
+
+    public void setEnglishLearners(final String englishLearners) {
+        this.englishLearners = englishLearners;
+    }
+
+    public boolean isLepEnabled() {
```
Comment (esotrope) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettings.java]: i know we abbr this but should we? comments should at least expand this text for us

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettings.java (hunk 2 above)
Comment (esotrope) [reply common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettings.java]: wondering if this is more appropriately a collection? and we test enabled with collection.contains

File: webapp/src/main/webapp/src/app/assessments/filters/adv-filters/adv-filters.component.html [hunk 4]
```diff
@@ -5,7 +5,7 @@
 <div class="flex-children">
 
   <!-- Test Filters -->
-  <div id="filter-by-test" class="flex-child">
+  <div id="filter-by-test" class="flex-child no-margin">
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/filters/adv-filters/adv-filters.component.html]: this a bootstrap class or ui kit class? if ui-kit i'm surprised we didn't call it `m-0` to go with `mX-0`

File: webapp/src/main/webapp/src/app/assessments/model/filter-by.model.ts [hunk 5]
```diff
@@ -54,6 +59,11 @@ export class FilterBy extends ObservableObject {
         for (let filteredGender of filteredGenders) {
           all.push(property + "." + filteredGender);
         }
+      } else if (property == "elasCodes") {
+        const filteredElasCodes = this.filteredElasCodes;
+        for (let i of filteredElasCodes) {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/model/filter-by.model.ts]: i should be reserved for representing `index` also when given the op - change the `let` to `const` when the value isn't updated in the block

File: webapp/src/main/webapp/src/assets/i18n/en.json [hunk 6]
```diff
@@ -410,6 +410,13 @@
       "Section504": "504",
       "StudentEnrolledGrade": "Student Enrolled Grade"
     },
+    "elas": {
+      "EO": "English Only",
+      "EL": "English Learner",
+      "IFEP": "Initial Fluent English Proficient",
+      "RFEP": "Reclassified Fluent English Proficient",
+      "TBD": "To Be Determined"
+    },
```
Comment (esotrope) [inline webapp/src/main/webapp/src/assets/i18n/en.json]: this going to be something we disaggregate on in aggregates?

File: webapp/src/main/webapp/src/app/assessments/filters/adv-filters/adv-filters.component.html (hunk 4 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/assessments/filters/adv-filters/adv-filters.component.html]: its in the ui kit. Looking around it looks like there are a bunch is a ml-0, mr-0, mt-0, mb-0, but I don't see anything for all margins

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettings.java (hunk 3 above)
Comment (dsnyder-fairway) [reply common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettings.java]: removed abbr and added some javadoc

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettings.java (hunk 2 above)
Comment (dsnyder-fairway) [reply common/src/main/java/org/opentestsystem/rdw/reporting/common/configuration/ReportingSystemSettings.java]: good call. Changed

File: webapp/src/main/webapp/src/assets/i18n/en.json (hunk 6 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/assets/i18n/en.json]: I believe so..

**Pull request merged**